music21/stream/__init__.py

```diff
@@ -3972,18 +3973,12 @@ def hasMeasureNumberInformation(measureIterator):
             mOffset = m.getOffsetBySite(srcObj) - startOffset
             returnObj.coreInsert(mOffset, m)
 
-        if gatherSpanners:
-            sf = srcObj.flat
-            for sp in spannerBundle:
-                # can use old offsets of spanners, even though components
-                # have been updated
-                # returnObj.insert(sp.getOffsetBySite(mStreamSpanners), sp)
-                returnObj.coreInsert(sf.elementOffset(sp), sp)
-
-                # environLocal.printDebug(['Stream.measures: copying spanners:', sp])
-
         # used coreInsert
         returnObj.coreElementsChanged()
+
+        if gatherSpanners:
+            returnObj.coreGatherMissingSpanners()
```

Comment:
Oh, hey!  This was already here.  How nice!  :-D  wow.

I wonder if the default should be not to `requireAllPresent` so that we can extract measure 3-4 and see "are there any slurs beginning or ending in mm. 3-4?"  It'd be a lighter transition from v6 if at least any possibly relevant spanner was there.  We could make "gatherSpanners" be an enum (another PR adds a common.enums file that could go in.) of True, False, or CompleteOnly  -- feel free to leave that for another PR.

I'll go and update coreGatherMissingSpanners() to be more efficient.

Reply:
Nice, agreed that that would be a lighter transition.